tests/test_shellsyntax.py

```diff
@@ -394,33 +394,243 @@ def setUp(self):
+        self.rpm.add_installed_file('/usr/share/data/valid_bash.sh',
+                                    rpmfluff.SourceFile('valid_bash.sh', valid_bash))
+        self.inspection = 'shellsyntax'
+        self.label = 'shell-syntax'
+        self.result = 'OK'
 
-# Valid /bin/bash script is OK for comparing Koji builds
 
-# Invalid /bin/bash script is BAD for comparing Koji builds
+class BashMalformedRPM(TestRPMs):
+    """
+    Invalid /bin/bash script is BAD for RPMs
+    """
+    def setUp(self):
+        TestRPMs.setUp(self)
 
-# Valid but requires 'extglob' script is OK for comparing
-# Koji builds
+        self.rpm.add_installed_file('/usr/share/data/invalid_bash.sh',
+                                    rpmfluff.SourceFile('invalid_bash.sh', invalid_bash))
+        self.inspection = 'shellsyntax'
+        self.label = 'shell-syntax'
+        self.result = 'BAD'
+        self.waiver_auth = 'Anyone'
 
-# Valid /bin/bash script in before, invalid in after is BAD when
-# comparing RPMs
 
-# Valid /bin/bash script in before, invalid in after is BAD when
-# comparing Koji builds
+class BashExtglobWellFormedRPM(TestRPMs):
+    """
+    Valid but requires 'extglob' script is OK for RPMs
+    """
+    def setUp(self):
+        TestRPMs.setUp(self)
+
+        self.rpm.add_installed_file('/usr/share/data/valid_bash_extglob.sh',
+                                    rpmfluff.SourceFile('valid_bash_extglob.sh', valid_bash_extglob))
+        self.inspection = 'shellsyntax'
+        self.label = 'shell-syntax'
+        self.result = 'INFO'
+
+
+class BashWellFormedKoji(TestKoji):
+    """
+    Valid /bin/bash script is OK for Koji build
+    """
+    def setUp(self):
+        TestKoji.setUp(self)
+
+        self.rpm.add_installed_file('/usr/share/data/valid_bash.sh',
+                                    rpmfluff.SourceFile('valid_bash.sh', valid_bash))
+        self.inspection = 'shellsyntax'
+        self.label = 'shell-syntax'
+        self.result = 'OK'
+
+
+class BashMalformedKoji(TestKoji):
+    """
+    Invalid /bin/bash script is BAD for Koji build
+    """
+    def setUp(self):
+        TestKoji.setUp(self)
+
+        self.rpm.add_installed_file('/usr/share/data/invalid_bash.sh',
+                                    rpmfluff.SourceFile('invalid_bash.sh', invalid_bash))
+        self.inspection = 'shellsyntax'
+        self.label = 'shell-syntax'
+        self.result = 'BAD'
+        self.waiver_auth = 'Anyone'
+
+
+class BashExtglobWellFormedKoji(TestKoji):
+    """
+    Valid but requires 'extglob' script is OK for Koji build
+    """
+    def setUp(self):
+        TestKoji.setUp(self)
+
+        self.rpm.add_installed_file('/usr/share/data/valid_bash_extglob.sh',
+                                    rpmfluff.SourceFile('valid_bash_extglob.sh', valid_bash_extglob))
+        self.inspection = 'shellsyntax'
+        self.label = 'shell-syntax'
+        self.result = 'INFO'
+
+
+class BahWellFormedCompareRPMs(TestCompareRPMs):
```

Comment:
"Bash" rather than "Bah", presumably?
```suggestion
class BashWellFormedCompareRPMs(TestCompareRPMs):
```